src/vm/instruction.ts

```diff
@@ -27,40 +27,130 @@ import {
 // └─────┴─────┴───┴───┴───┴───┴───┴───┴───┴───┴────┴────┴────┴────┴────┴────┘
 // Source: https://github.com/lambdaclass/cairo-vm_in_go/blob/main/pkg/vm/instruction.go
 
-// Dst & Op0 register flags
-export type RegisterFlag = 'ap' | 'fp';
+/** Flags to compute `dst` and `op0` from a register */
```

Comment:
I wouldn't put a contextual description as a comment. Here you describe the register in the context of it's usage in an instruction decoding.

Just present the register as a pure definition, and add `pc`

Reply:
This enum is only used to represent the value which can be taken by `dst_reg` and `op0_reg`, which are either `ap` or `fp`, to compute the value of `dst` and `op0`. I don't think that `pc` has its place here. (See '# compute op0' and '# compute dst' in the State Transition pseudo-code of the cairo whitepaper p.33, p.34)

Knowing this context, should the register still be described with pure description (what each pointer is) or a contextual description ?

Reply:
If the enum was to represent the value taken in the encoded instruction, then you'd have here the `0` and `1`, and not hardcoded below (which is fine for me)

```
    let dstRegister: Register;
    if (targetdstRegister == 0n) {
      dstRegister = Register.Ap;
    } else {
      dstRegister = Register.Fp;
    }
```

there is no reason for me not to have all the registers in this enum